spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -16,4 +16,32 @@ def foobar_action
       get :foobar_action
     end
   end
+
+  describe 'while signed-in' do
+    before :each do
+      sign_in instance_double(AdminUser)
+    end
+
+    describe 'GET #show' do
+      before :each do
+        @id= FactoryGirl.create(:partner).id
+      end
+
+      describe 'redirect & flash' do
+        example 'with invalid :id' do
+          get :show, id: 0
```

Comment:
using `''` will give a path without an id so `params[:id]` will not exist.  (/partners/show/)
But if the tests are go then it's gooone! :)